### ISSUE TITLE
error loading dropdown

### ISSUE THREAD

**Issue by dmfrey** [thecount2a/mythtv-stream-mpeg-dash#3]
Getting this in apache error log

    [Thu Feb 16 19:31:23.489761 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 59
    [Thu Feb 16 19:31:23.489993 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined offset: 0 in /var/www/html/dash/index.php on line 70
    [Thu Feb 16 19:31:23.494272 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Warning:  mysql_fetch_assoc() expects parameter 1 to be resource, boolean given in /var/www/html/dash/index.php on line 90
    [Thu Feb 16 19:31:23.494310 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined offset: 0 in /var/www/html/dash/index.php on line 107
    [Thu Feb 16 19:31:23.494327 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 118
    [Thu Feb 16 19:31:23.494345 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 118
    [Thu Feb 16 19:31:23.494354 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 120
    [Thu Feb 16 19:31:23.494360 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined index: action in /var/www/html/dash/index.php on line 121
    [Thu Feb 16 19:31:23.494367 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined index: action in /var/www/html/dash/index.php on line 148
    [Thu Feb 16 19:31:23.494374 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined index: action in /var/www/html/dash/index.php on line 174
    [Thu Feb 16 19:31:23.494388 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 552
    [Thu Feb 16 19:31:23.494402 2017] [:error] [pid 26936] [client 192.168.100.1:55203] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 650

Looks like it is trying access the `filename` request parameter. Tried supplying it with no, but then it throws another exception.

**Comment by dmfrey** [comment]
anything on this?

**Comment by thecount2a** [comment]
Sorry, I've been out of town for a week for a conference so I haven't been
checking all of my emails.

Can you describe the symptoms you are seeing?  I only see "Notice" and
"Warning" which just means that my code does not check the existence of
some variables before testing their value which is bad practice in PHP but
not fatal.  Can you tell me what index.php shows you when you try to set
filename=XXXX_XXXXXXXXXXXXXX where the Xs are replaced with the filename of
the file you want to view (with the .mpg extension removed).  If your
mythtv does not terminate recording filenames with .mpg, I may need to fix
the code to properly deal with that scenario.

Thanks.

**Comment by dmfrey** [comment]
I am out of town now and will check this when I'm home. But I think they
are .ts files.

**Comment by thecount2a** [comment]
This could be the problem then.  My version of mythtv saves the recordings
(which are actually transport stream files) as .mpg.  I do remember reading
that some later versions of mythtv correctly save them as .ts files.  There
are 6 instances in index.php where the .mpg extension is used so you should
be able to try changing the code to use .ts instead of .mpg to see if that
fixes everything.

If this is the case, I can try to do some logic at the top of the program
to determine what extension is being used and adjust the subsequent
references accordingly.

**Comment by dmfrey** [comment]
I modified the file and the `.ts` does seem to have fixed some of it. I ran into some other things like permissions that I thought were set on the `videos` directory. Also, I had copied in the uncompiled shaka player instead of the compiled on.  It still is erroring out.  I see a number of messages in the apache2 error log.

    [Mon Feb 27 12:08:31.980068 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined offset: 0 in /var/www/html/dash/index.php on line 70
    [Mon Feb 27 12:08:31.980924 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined offset: 0 in /var/www/html/dash/index.php on line 107
    [Mon Feb 27 12:08:31.981033 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 118
    [Mon Feb 27 12:08:31.981071 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 118
    [Mon Feb 27 12:08:31.981096 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 120
    [Mon Feb 27 12:08:31.981113 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined index: action in /var/www/html/dash/index.php on line 121
    [Mon Feb 27 12:08:31.981129 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined index: action in /var/www/html/dash/index.php on line 148
    [Mon Feb 27 12:08:31.981144 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined index: action in /var/www/html/dash/index.php on line 174
    [Mon Feb 27 12:08:31.981170 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 552
    [Mon Feb 27 12:08:31.981198 2017] [:error] [pid 31710] [client 192.168.100.114:60144] PHP Notice:  Undefined index: filename in /var/www/html/dash/index.php on line 650

**Comment by thecount2a** [comment]
Again, all of these are "PHP Notice" messages which are unfortunately expected due to my bad practice of checking array indexes that may legitimately not exist.

What are you seeing on the page after you access index.php?filename=<YOUR FILENAME WITHOUT EXTENSION>?  Do you get the form to specify encoding quality?  If so, what happens after you submit that form?

**Comment by dmfrey** [comment]
yes, i get the page when i supply a filename. the player never seems to be
able to play anything. I do, however, see all the files in the videos
directory.

**Comment by thecount2a** [comment]
Can you look inside one of the directories in the videos directory and "cat" the contents of status.txt, packager.log and encode.log?  The contents of these files may reveal what is going wrong and why it doesn't allow you to begin viewing videos.

**Comment by dmfrey** [comment]
Status showed completed successfully.

I then looked in chrome inspector and saw this

    Uncaught ReferenceError: goog is not defined
        at shaka-player.uncompiled.js:23

So something isn't right with the shaka player yet

**Comment by thecount2a** [comment]
Are you using the shaka-player.compiled.js file exactly as it comes from the following URL?

https://cdnjs.cloudflare.com/ajax/libs/shaka-player/2.0.1/shaka-player.compiled.js

**Comment by dmfrey** [comment]
I replaced it with the one from the link you posted. I no longer see the
warning in chrome inspector.  However, the video still does not play. The
controls are kind of grayed out.  It does report processing is complete at
100% now.

**Comment by thecount2a** [comment]
It would be nice to see the actual contents of the various log files that I previously mentioned.  My guess would be that there might be some error in the "encode.log" file or the "packager.log" file.  Also, it would be good to know if the "status.txt" file ever mentions that the remux has finished successfully.  Could you copy the contents of the "status.txt" file verbatim for me to analyze?  Thanks.  

One more thing to try. You could also try clicking the "Delete Video Files" button and then relaunch the process again (by again accessing index.php?filename=YOUR_FILENAME)

**Comment by thecount2a** [comment]
Thanks for being the first to test this out, I will probably modify the index.php to deal with multiple transport stream file extensions and create a quick install script to make this easier for people in the future.

Any status on your setup?  Did you get videos playing back through the shaka player?
